instagram/src/components/Authentication/withAuthenticate.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from "react";
+
+const withAuthenticate = AppContainer => Login =>
+  class extends Component {
+    state = {
+      auth: false,
+      username: "",
+      password: ""
+    };
+
+    componentDidMount() {
+      if (JSON.parse(localStorage.getItem("auth", "true"))) {
+        this.setState({
+          auth: true
+        });
+      }
+    }
+
+    login = e => {
+      e.preventDefault();
+      localStorage.setItem("auth", "true");
+      this.setState({ auth: !this.state.auth });
+    };
+
+    logout = () => {
+      localStorage.removeItem("auth");
+      this.setState({ auth: !this.state.auth });
+    };
+
+    getUsername = value => {
```

Comment:
This is a smart way to get username from the get go, and then pass it down to the components that need it. You could've also just gotten that username from local storage wherever you needed it.